// nuxt3-leaflet/dist/module.mjs
import { defineNuxtModule, addPlugin, addComponent } from '@nuxt/kit';
import path from 'path';



// -- Unbuild CommonJS Shims --
import __cjs_url__ from 'url';
import __cjs_path__ from 'path';
import __cjs_mod__ from 'module';
const __filename = __cjs_url__.fileURLToPath(import.meta.url);
const __dirname = __cjs_path__.dirname(__filename);
const require = __cjs_mod__.createRequire(import.meta.url);
const components = [
  "LCircle",
  "LCircleMarker",
  "LControl",
  "LControlAttribution",
  "LControlLayers",
  "LControlScale",
  "LControlZoom",
  "LFeatureGroup",
  "LGeoJson",
  "LIcon",
  "LImageOverlay",
  "LLayerGroup",
  "LMap",
  "LMarker",
  "LPolygon",
  "LPolyline",
  "LPopup",
  "LRectangle",
  "LTileLayer",
  "LTooltip",
  "LWmsTileLayer"
];
const module = defineNuxtModule({
  meta: {
    name: "nuxt-leaflet",
    configKey: "nuxtLeaflet",
    compatibility: {
      nuxt: "^3.0.0"
    }
  },
  // Default configuration options of the Nuxt module
  defaults: {},
  async setup(options, nuxt) {
    nuxt.options.css.push("leaflet/dist/leaflet.css");
    addPlugin({
      src: path.resolve(__dirname, "runtime/leaflet-runtime"),
      mode: "client"
    });
    for (const component of components) {
      addComponent({
        name: component,
        export: component,
        filePath: "@vue-leaflet/vue-leaflet",
        chunkName: `nuxt-leaflet/${component}`,
        mode: "all"
      });
    }
  }
});

export { components, module as default };
